code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -78,6 +81,77 @@
 	if(world.time <= next_move)
 		return
 	A.attack_ghost(src)
+	if(ishuman(A) && in_range(src, A))
+		Harvest(A, src)
+
+
+/mob/living/simple_animal/revenant/proc/Harvest(mob/living/carbon/human/tar, mob/living/simple_animal/revenant/user = usr)
+	if(!user.castcheck(0))
+		return
+	var/mob/living/carbon/human/target = tar
```

Comment:
why. ```tar``` is already human, just do ```if(istype(tar))``` and then you can use ```tar``` from then on.
if you don't like it being called ```tar``` (i don't) then change it's name in the arguments for ```Harvest()```